tests/test_pytest_cov.py

```diff
@@ -357,6 +358,56 @@ def test_cov_min_float_value(testdir):
     ])
 
 
+def test_cov_min_multi(testdir):
+    module = testdir.mkpydir("legacy")
+    module.join("uncovered.py").write(
+        textwrap.dedent(
+            """\
+            import sut
+
+            def some_legacy_code():
+                strategy = sut.AbstractFactoryBean()
+                assert strategy.business_critical()
+            """
+        )
+    )
+    module.join("covered.py").write(
+        textwrap.dedent(
+            """\
+            def two_returner_strategy():
+                return 2
+            """
+        )
+    )
+    testsubdir = testdir.mkdir("test")
+
+    testsubdir.join("test_legacy.py").write(
+        textwrap.dedent(
+            """\
+            from legacy import covered
+
+            def test_two_returner_strategy():
+                assert covered.two_returner_strategy() == 2
+            """
+        )
+    )
+    result = testdir.runpytest('-v',
+                               '--cov=%s' % str(testdir.tmpdir),
+                               '--cov-report=term-missing',
+                               '--cov-fail-under=55.55',
+                               '--cov-fail-under=100.0:test/*',
+                               '--cov-fail-under=100.0:+test/*',
+                               '--cov-fail-under=33.33:-test/*',
+                               testsubdir)
+    assert result.ret == 0
+    result.stdout.fnmatch_lines([
+        "Required test coverage of 55.55% reached. Total coverage: 55.56%",
+        "Required test coverage of 100.0%:+test/* reached. Total coverage: 100.00%",
+        "Required test coverage of 100.0%:+test/* reached. Total coverage: 100.00%",
+        "Required test coverage of 33.33%:-test/* reached. Total coverage: 33.33%",
+    ])
```

Comment:
So to understand what the argument do - in this test it's like this right?
- overall coverage must be at least 55.55%
- test coverage must be 100%
- test coverage must be 100% (alternative syntax with "+")
- overall coverage but without tests must be at least 33.33%

Also if I understand it correctly "%" is optional, and "+" as well yes?

And `--cov-fail-under=100:foo:bar` means coverage must be 100% for `foo` **and** `bar` right? It should be part of the test suite.

There are few things that I think users will find confusing or produce undesired results:
- the multiple syntax to obtain the same thing
- the reporting discrepancy - you can set fail-under for something that you cannot see in the reporting - suddenly you can't know what to fix to make the suite pass

Do we really need the omit mode (the "-" syntax)?

Reply:
> So to understand what the argument do - in this test it's like this right?
> 
>     * overall coverage must be at least 55.55%
> 
>     * test coverage must be 100%
> 
>     * test coverage must be 100% (alternative syntax with "+")
> 
>     * overall coverage but without tests must be at least 33.33%
> 
> 
> Also if I understand it correctly "%" is optional, and "+" as well yes?
> 
> And `--cov-fail-under=100:foo:bar` means coverage must be 100% for `foo` **and** `bar` right? It should be part of the test suite.
> 
> There are few things that I think users will find confusing or produce undesired results:
> 
>     * the multiple syntax to obtain the same thing
I don't think it's confusing. You get the same with +1 and 1
> 
>     * the reporting discrepancy - you can set fail-under for something that you cannot see in the reporting - suddenly you can't know what to fix to make the suite pass
I don't believe you can
> 
> 
> Do we really need the omit mode (the "-" syntax)?
Yes so you can get an idea of coverage for code without tests. This is the coverage result people ignore the test suite for.

Reply:
Most people upgrading to this syntax will have ignored tests globally and will still want to assert on that number

Reply:
> Yes so you can get an idea of coverage for code without tests. This is the coverage result people ignore the test suite for.

Most people have just one top level package thus omit is unnecessary.

> I don't believe you can

The reporting show percentages for individual files, doesn't show any totals for packages, subpackages and so on. This is why I don't think this is a great idea.

@nedbat do you have any input on this syntax? I don't want to have something in pytest-cov that will never ever be part of coveragepy.

Reply:
> > Yes so you can get an idea of coverage for code without tests. This is the coverage result people ignore the test suite for.
> 
> Most people have just one top level package thus omit is unnecessary
That doesn't follow:
Most people have just one top level package
thus some people do not have just one top level package
thus omit is necessary
> 
> > I don't believe you can
> 
> The reporting show percentages for individual files, doesn't show any totals for packages, subpackages and so on. This is why I don't think this is a great idea.
No the reporting lists a percentage for each --cov-fail-under

> @nedbat do you have any input on this syntax? I don't want to have something in pytest-cov that will never ever be part of coveragepy.
This interface is already available in coveragepy: you just run "coverage report" with different include/omit overrides